waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/TSetIpAddressProcessorFactory.java

```diff
@@ -47,6 +53,9 @@ public TSetIpAddressProcessorFactory(
   @Override
   public TProcessor getProcessor(TTransport transport) {
     try {
+      Socket socket = ((TSocket) transport).getSocket();
```

Comment:
I *think* the way we have this all set up this will always be of type `TSocket` so this should never throw a `ClassCastException` right? Alternative would be to do an `instanceof` check first...

Reply:
yes, I didn't bother with the check. Should be alright.